auth/provider/gitlab_profile.go

```diff
@@ -0,0 +1,79 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// GitLabProvider provider
+type GitLabProvider struct {
+	BaseURL string
+	ApiURL  string
+	Client  *http.Client
+}
+
+// NewGitLabProvider create a new GitLabProvider API provider
+func NewGitLabProvider(c *http.Client, baseURL string, apiURL string) *GitLabProvider {
+	return &GitLabProvider{
+		Client:  c,
+		BaseURL: baseURL,
+		ApiURL:  apiURL,
+	}
+}
+
+// GetProfile returns a profile for a user from GitLabProvider
+func (gl *GitLabProvider) GetProfile(accessToken string) (*Profile, error) {
+	var err error
+	var gitlabProfile GitLabProfile
+
+	req, reqErr := http.NewRequest(http.MethodGet, gl.ApiURL + "user", nil)
+	req.Header.Add("Authorization", "bearer "+accessToken)
+
+	if reqErr != nil {
+		return nil, reqErr
+	}
+
+	res, err := gl.Client.Do(req)
+
+	if err != nil {
+		return nil, reqErr
```

Comment:
I think you want to return `err` rather than `reqErr` here since `reqErr` is caught and returned just above.

Reply:
Another good catch! Thank you! :-)